### ISSUE TITLE
Mattermost Notifications not working if using IP:Port

### ISSUE THREAD

**Issue by anonaut** [Tautulli/Tautulli#955]
**Version:** 
1.4.16
**Branch:** 
master
**Commit hash:** 	
3742f33d08feb6fb15e74af6318185230205006f
**Operating system:** 
Ubuntu 16.04
**Python version:** 
2.7.6
**What you did?** 
Installed mattermost and tried to switch the notification settings from slack to mattermost.
**What happened?**
When i send the testmessage i got a red popup "cherrypy 500 internal error", when i saved the settings i get an uncaught exception when a notification is sent.
**What you expected?**
To receive the notification in mattermost.
**How can we reproduce your issue?**

1. Insert Mattermost Incoming Hook Url
2. Press "Test Slack"

My mattermost install doesn't have a domain so i tried it with ip:port/webhookurl - after it didn't work i created a local hostname for the ip and tried again - still doesn't work. I thought maybe it doesn't like ports in the url so i configured my nginx on the mattermost machine to reverse proxy it and tried again - still doesn't work.

I can create a test notification from the plexpy hostmachine with curl, so it can't be the hostname/reverse proxy - i also tried the webhook with sonarr - it works here with ip:port/webhookurl schema.

For any help i would be glad. I can supply a mattermost webhook if needed to test it.

I don't post the complete log because there is nothing else except those errors.

```
2017-01-30 21:43:55 - ERROR   :: Thread-53 : Uncaught exception: Traceback (most recent call last):
  File "/opt/plexpy/plexpy/logger.py", line 270, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/opt/plexpy/plexpy/notification_handler.py", line 176, in notify
    metadata=metadata)
  File "/opt/plexpy/plexpy/notifiers.py", line 599, in send_notification
    return slackClient.notify(message=body, event=subject, **kwargs)
  File "/opt/plexpy/plexpy/notifiers.py", line 2071, in notify
    body=json.dumps(data))
  File "/usr/lib/python2.7/httplib.py", line 1017, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python2.7/httplib.py", line 1051, in _send_request
    self.endheaders(body)
  File "/usr/lib/python2.7/httplib.py", line 1013, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 864, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 826, in send
    self.connect()
  File "/usr/lib/python2.7/httplib.py", line 1216, in connect
    self.timeout, self.source_address)
  File "/usr/lib/python2.7/socket.py", line 571, in create_connection
    raise err
error: [Errno 111] Connection refused
```

**Comment by JonnyWong16** [comment]
I can see a port number causing an issue, but I don't know why a domain wouldn't work. Is it the same error when you try with a domain instead? Is your webhook url using HTTPS?

Edit: Basically, as it is coded now, it needs to be running with HTTPS on port 443.

**Comment by anonaut** [comment]
Thanks for the fast reply.

I debugged it myself more and jumped into the python interpreter after sometime and tried to fetch the webpage of my hostname and as it turns out python doesn't care about my /etc/hosts entry. 

I don't really know how python resolves domain names (thats something to learn tomorrow), but i think we can close this because it will probably work with a real domain.

**Comment by JonnyWong16** [comment]
I think I will still label this as a bug because it should consider the port number if you are using it for 
Mattermost.

A Slack webhook will always be on port 443, so I never considered that.

**Comment by JonnyWong16** [comment]
Fixed in v1.4.20.
